presto-native-execution/presto_cpp/main/operators/LocalPersistentShuffle.cpp

```diff
@@ -168,16 +165,61 @@ BufferPtr TestingPersistentShuffle::next(int32_t partition, bool success) {
   return buffer;
 }
 
-bool TestingPersistentShuffle::readyForRead() const {
+bool LocalPersistentShuffle::readyForRead() const {
   auto readyFile = fileSystem_->openFileForRead(
       fmt::format("{}/{}", rootPath_, kReadyForReadFilename));
   return (readyFile != nullptr);
 }
 
-void TestingPersistentShuffle::cleanup() {
+void LocalPersistentShuffle::cleanup() {
   auto files = fileSystem_->list(rootPath_);
   for (auto& file : files) {
     fileSystem_->remove(file);
   }
 }
-} // namespace facebook::presto::operators
\ No newline at end of file
+
+using json = nlohmann::json;
+
+// static
+LocalShuffleReadInfo LocalShuffleInfoDeserializer::deserializeReadInfo(
+    const std::string& info) {
+  auto jsonReadInfo = json::parse(info);
+  LocalShuffleReadInfo readInfo;
+  jsonReadInfo.at("rootPath").get_to(readInfo.rootPath);
+  jsonReadInfo.at("numPartitions").get_to(readInfo.numPartitions);
+  return readInfo;
+}
+
+// static
+LocalShuffleWriteInfo LocalShuffleInfoDeserializer::deserializeWriteInfo(
+    const std::string& info) {
+  auto jsonWriteInfo = json::parse(info);
+  LocalShuffleWriteInfo writeInfo;
+  jsonWriteInfo.at("rootPath").get_to(writeInfo.rootPath);
+  jsonWriteInfo.at("numPartitions").get_to(writeInfo.numPartitions);
+  return writeInfo;
+}
+
+// static
+std::shared_ptr<ShuffleInterface> LocalPersistentShuffle::create(
+    const std::string& serializedShuffleInfo,
+    operators::ShuffleInterface::Type type,
+    velox::memory::MemoryPool* pool) {
+  if (type == operators::ShuffleInterface::Type::kRead) {
+    operators::LocalShuffleReadInfo readInfo =
+        operators::LocalShuffleInfoDeserializer::deserializeReadInfo(
+            serializedShuffleInfo);
+    return std::make_shared<operators::LocalPersistentShuffle>(
+        readInfo.rootPath, pool, readInfo.numPartitions, 1 << 15);
```

Comment:
Shall we define "1 << 15" as a static const class and make a query config in followup?